server/validate.js

```diff
@@ -16,17 +20,79 @@ exports.validateRegister = [
     if (!errors.isEmpty())
       return res.status(400).json({ errors: errors.array() });
     next();
-  }
+  },
 ];
 
 exports.validateLogin = [
-  check("email", "Please enter a valid email address").isEmail(),
-  check("password", "Password is required").not().isEmpty(),
+  check("email", `${msgPrefix} email address`).isEmail(),
+  check("password", "Password is required").notEmpty(),
+  (req, res, next) => {
+    const errors = validationResult(req);
+
+    if (!errors.isEmpty())
+      return res.status(400).json({ errors: errors.array() });
+    next();
+  },
+];
+
+exports.validateProfileDetails = [
+  check("firstName", ` ${msgPrefix} first name`)
+    .exists(checkFalsyVal)
+    .trim()
+    .escape()
+    .isLength({ max: 20 }),
+  check("lastName", ` ${msgPrefix} last name`)
```

Comment:
Good job! Really great to see middleware validation being added, it adds that extra reliability onto our API

Reply:
The concept is well understood.